pkg/collector/checkhelpers/backoff/backoff.go

```diff
@@ -0,0 +1,167 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package backoff
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"sync"
+)
+
+const (
+	countdown = "countdown"
+	maximum   = "maximum"
+)
+
+type RetryableOperation[T any] func() (T, error)
+
+type Store struct {
+	lock       sync.RWMutex
+	strategies map[string]map[string]int // "operation_name": { "countdown": X, "maximum": Y }
+}
+
+func New() *Store {
+	return &Store{
+		strategies: make(map[string]map[string]int),
+	}
+}
+
+func (s *Store) Delete(name string) {
+	s.lock.Lock()
+	delete(s.strategies, name)
+	s.lock.Unlock()
+}
+
+// Get returns the strategy's countdown, i.e. how many checks to skip until retrying
+func (s *Store) Get(name string) (int, bool) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	// Get the backoff strategy ex: "operation_name": { "countdown": 2, "maximum": 4 }
+	strategy, ok := s.strategies[name]
+	if !ok {
+		return 0, false
+	}
+
+	// Check that the strategy's countdown key exists
+	status, ok := strategy["countdown"]
+	if !ok {
+		return 0, false
+	}
+
+	return status, ok
+}
+
+// Init initializes both the "countdown" and "maximum" keys associated with an operation's retries
+// We need to track two fields in the map because we need to:
+// - keep track of the maximum backoff reached (so we can exponentially backoff)
+// - keep track of the current strategy's countdown (so we can keep track of how many iterations have elapsed)
+func (s *Store) Init(name string, init int) bool {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	// Guard against initializing a key that already exists
+	if _, ok := s.strategies[name]; ok {
+		return ok
+	}
+
+	// Initialize a new strategy map with given value
+	s.strategies[name] = map[string]int{countdown: init, maximum: init}
+
+	return true
+}
+
+// Decrement decreases the strategy's countdown by 1, called each time an attempt is skipped
+func (s *Store) Decrement(name string) bool {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	// Get the backoff strategy e.g. "operation_name": { "countdown": 2, "maximum": 4 }
+	strategy, ok := s.strategies[name]
+	if !ok {
+		return false
```

Comment:
All these return values seem to be unused (here and on all other map accesses). Why are they necessary?